### DISCUSSION TITLE
How to group some parts of select primitives in my custom components and use it?

### DISCUSSION THREAD

**Discussion by Lirrrx** [radix-ui/primitives#1714] [Help] [answered]
Hello there. I'm trying to group some parts of select primitives in my custom components and use it.

https://codesandbox.io/s/react-tailwind-starter-forked-cwmw6p?file=/src/Select.js:1974-2168

I created 2 versions of Select in files: Select.js and MySelect.js. Select.js component works fine because i use code from example, but MySelect works incorrect. How can i rewrite <Select /> and <SelectItem /> components to make it work.

**Comment by joaom00** [comment, marked as answer]
Hi @Lir-kulikov,

You forgot to pass `value` prop to `SelectPrimitive.Item` in `MySelectItem`.

Additionally, they [have an example](https://www.radix-ui.com/docs/primitives/components/select#custom-apis) to create your own API.

**Reply by Lirrrx** [reply]
Thanks a lot! i completely missed this part of documentation.
